### PR TITLE
Make app async

### PR DESCRIPTION
**Pull request by ljiatu**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Unfortunately, had to use the v6.0.0-beta9 version of the `web3` package to get native async support.

* **What is the current behavior?** (You can also link to an open issue here)
All network requests are synchronous.

* **What is the new behavior (if this is a feature change)?**
All network requests are asynchronous.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No

* **Other information**:
~Need to make file I/O async.~ File I/O is also async now.

### CODE REVIEW

**Comment by PlayJok3r** [comment]
The change looks good, but the linter says import order needs to be adjusted

**Review by R0rschach — APPROVED** [review]
lgtm!

File: pyproject.toml [hunk 1]
```diff
@@ -14,8 +14,11 @@ structlog = "^22.3.0"
 uvicorn = "^0.20.0"
 pandas = "^1.5.2"
 python-dotenv = "^0.21.0"
-web3 = "^5.31.3"
+web3 = {version = "^6.0.0-beta9", allow-prereleases = true}
 sentry-sdk = {extras = ["fastapi"], version = "^1.13.0"}
+httpx = "^0.23.3"
+base58 = "^2.1.1"
+ipfshttpclient = "^0.7.0"
```
[the remark this answers is not in the document]
Comment (ljiatu) [reply pyproject.toml:21]: Yes, although not explicitly. It's needed for the updated web3 package.

File: pyproject.toml [hunk 2]
```diff
@@ -14,8 +14,11 @@ structlog = "^22.3.0"
 uvicorn = "^0.20.0"
 pandas = "^1.5.2"
 python-dotenv = "^0.21.0"
-web3 = "^5.31.3"
+web3 = {version = "^6.0.0-beta9", allow-prereleases = true}
```
[the remark this answers is not in the document]
Comment (ljiatu) [reply pyproject.toml:17]: Unfortunately, had to use the v6.0.0-beta9 version of the web3 package to get native async support.

**Pull request merged**